Fix 697281: Fixes response error code from 500 to 400

asmithakarun opened the pull request:
The response error code for invalid arguments has been changed from 500 to 400.

Tested By:
curl -X PATCH https://rain111bmc.aus.stglabs.ibm.com:18080/redfish/v1/Systems/system/Bios/Settings -d '{"Attributes": {"pvm_stop_at_standby": "false"}}' Response
HTTP/1.1 400 Bad Request

abhilashraju committed:
Fix 697281: Fixes response error code from 500 to 400

The response error code for invalid arguments has been changed from 500
to 400.

Tested By:
curl -X PATCH https://rain111bmc.aus.stglabs.ibm.com:18080/redfish/v1/Systems/system/Bios/Settings -d '{"Attributes": {"pvm_stop_at_standby": "false"}}'
Response
HTTP/1.1 400 Bad Request

Signed-off-by: Abhilash Raju <abhilash.kollam@gmail.com>

baemyung commented on redfish-core/lib/bios.hpp:
```diff
@@ -635,13 +635,36 @@ inline void handleBiosSettingsPatch(
+                                    pendingAttributesJson);
+                                return;
+                            }
+                        }
```
Missing `internalError()` here (& and better to have LOG_ERROR).

```
            BMCWEB_LOG_ERROR("BUS response error: {}", ec1);
            messages::internalError(asyncResp->res);
            return;
```

asmithakarun committed:
Use Bios dbus util and Add Bios ValueOutOfRange

Current implementation displays internal error message when BIOS
attribute value is not within the lower and upper bound values.
Also, the same internal error message is displayed when the length
of the String type value for the attribute to be updated is not
within the specified limit.

This commit displays property value out of range error message for
Integer type-based attributes in case of an error, and property value
incorrect for String type-based attributes.

Testing:
```
$ curl -k -H "Content-Type: application/json" -X PATCH \
    -d '{"Attributes":{"hb_mfg_flags_current": "000000000000000000000000000000000"}}' \
    https://${bmc}/redfish/v1/Systems/system/Bios/Settings
{
  "error": {
    "@Message.ExtendedInfo": [
      {
        "@odata.type": "#Message.v1_1_1.Message",
        "Message": "The property 'hb_mfg_flags_current' with the
requested value of '000000000000000000000000000000000' could not be
written because the value does not meet the constraints of the
implementation.",
        "MessageArgs": [
          "hb_mfg_flags_current",
          "000000000000000000000000000000000"
        ],
        "MessageId": "Base.1.8.1.PropertyValueIncorrect",
        "MessageSeverity": "Warning",
        "Resolution": "No resolution is required."
      }
    ],
    "code": "Base.1.8.1.PropertyValueIncorrect",
    "message": "The property 'hb_mfg_flags_current' with the requested
value of '000000000000000000000000000000000' could not be written
because the value does not meet the constraints of the implementation."
  }
}
```

```
$ curl -k -H "Content-Type: application/json" -X PATCH \
    -d '{"Attributes":{"hb_ioadapter_enlarged_capacity": 200}}' \
      https://${bmc}/redfish/v1/Systems/system/Bios/Settings
{
  "hb_ioadapter_enlarged_capacity@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "The value 200 for the property
hb_ioadapter_enlarged_capacity is not in the supported range of
acceptable values.",
      "MessageArgs": [
        "200",
        "hb_ioadapter_enlarged_capacity"
      ],
      "MessageId": "Base.1.13.0.PropertyValueOutOfRange",
      "MessageSeverity": "Warning",
      "Resolution": "Correct the value for the property in the request
body and resubmit the request if the operation failed."
    }
  ]
}
```

Signed-off-by: Asmitha Karunanithi <asmitk01@in.ibm.com>

baemyung reviewed (changes requested):
Please fix bios.hpp L770-794.

redfish-core-lib/bios.hpp:

```
                    if (ec1)
                    {
                        const sd_bus_error* dbusError = msg.get_error();
                        if (dbusError != nullptr)
                        {
                            ..
                        }

                        ---->>ADD HERE
                        BMCWEB_LOG_ERROR("BUS response error: {}", ec1);
                        messages::internalError(asyncResp->res); 
                         return;
                   }
```

baemyung reviewed (approved):
It looks good to me
